tests/cpp/plugin/test_federated_adapter.cu

```diff
@@ -54,7 +54,7 @@ TEST_F(FederatedAdapterTest, DeviceAllReduceSum) {
   for (auto rank = 0; rank < kWorldSize; rank++) {
     threads.emplace_back(std::thread([rank] {
       FederatedCommunicator comm{kWorldSize, rank, kServerAddress};
-      DeviceCommunicatorAdapter adapter{rank, &comm};
+      DeviceCommunicatorAdapter adapter{0, &comm};
```

Comment:
Is the correct?

Reply:
Using `rank` will error out because we now use only a single GPU for gtest.

Reply:
Got it, out of curiosity, after setting the rank to 0, is the test still testing what it's supposed to be testing?

Reply:
It still creates two threads, one for server and another for client

Reply:
Good to know. Please add a comment to it, otherwise, someone might try to "fix" it. ;-)